Runtime/Interaction/SnapZone.cs

```diff
@@ -317,16 +340,31 @@ public override void ProcessInteractor(XRInteractionUpdateOrder.UpdatePhase upda
             {
                 base.ProcessInteractor(updatePhase);
             }
-
+            
             if (socketActive)
             {
-                if (m_HoverTargets.Count == 0 && ShowHighlightObject)
+                hoverTargets.RemoveAll(target => target == null || target.enabled == false);
+                hoverTargets.First(target =>
```

Comment:
Is it guaranteed to have at least one element that matches this predicate? Otherwise, it will throw an exception. Also, isn't selectTarget can be checked only once instead for every element? Also, separate the lookup for an element from doing something over the element:
```
var hoverTarget = null;
if (selectTarget != null)
{
   hoverTarget = hoverTargets
    .Where(not selected)
    .Where(can select)
    .Intersect(m_hoverTargets) // Does the same as Where.Contains. Both are quite heavy.
    .FirstOrDefault();
}

if (hoverTraget != null)
{
    ForceSelect(hoverTarget);
}
```